### ISSUE TITLE
[BUG] Several issues with NLinear with `normalize=True`

### ISSUE THREAD

**Issue by turbotimon** [unit8co/darts#2699]
**Describe the bug**

After reading the paper and having a closer look at the implementation of `DLinearModel`, I see several issues:

1. The `seq_last` is only detached, but needs also to be cloned as well. Otherwise, it it's values will be changed by line https://github.com/unit8co/darts/blob/8bea0713494e6d23321feaaec341fff4e6fb33dc/darts/models/forecasting/nlinear.py#L149 . This is because of the tensors still share the same underlying memory. The solution is: `seq_last = x[:, -1:, : self.output_dim].detach().clone()`

2. Why is the normalization disabled by default? That's the whole catch of this model, so it should be enabled by default. (I could only imagine that it performed so bad due to the bug, that it was disabled by default)

3. The normalization is only done for the target and not covariates (https://github.com/unit8co/darts/blob/8bea0713494e6d23321feaaec341fff4e6fb33dc/darts/models/forecasting/nlinear.py#L145). How I understand it, the original implementation from the paper does it also for the covariates (https://github.com/cure-lab/LTSF-Linear/blob/0c113668a3b88c4c4ee586b8c5ec3e539c4de5a6/models/NLinear.py#L27). And in my opinion, this makes totally sense. For example, it is done also 

**To Reproduce**

With normalize=True, NLinear currently fails to solve even very simple tasks:

```
from darts import TimeSeries, models
from darts.utils.timeseries_generation import linear_timeseries
ts = linear_timeseries(start_value=0, end_value=10, length=10, freq='D')
model = models.NLinearModel(
    input_chunk_length=3,
    output_chunk_length=1,
    n_epochs=10,
    batch_size=1,
    const_init=True,
    normalize=True # Explore the difference when durning on normalize
    )
model.fit(ts)
print(model.predict(1))
```

**Expected behavior**

Output around ~10.0

**System (please complete the following information):**
 - Python version: 3.12
 - darts version 0.33.0

**Additional context**

**Comment by turbotimon** [comment]
Here are some debug screenshots to prove my claim (`seq_last` gets changed to 0):

![Image](https://github.com/user-attachments/assets/af58ac65-ec3d-42f1-851d-d4e88c7a9d5e)

![Image](https://github.com/user-attachments/assets/f1b0b853-f40f-4725-b087-d939093fa8a0)

**Comment by dennisbader** [comment]
> Agree. In addition, I think a warning:: in the model docstring for normalize would also be appropriate for it is a braking change.

True :) But I'd avoid having a warning raised everytime someone creates an `NLinearModel`, so a "🔴" (Breaking Change) note in the CHANGELOG.md should be enough.

> I see your point for the special case of future covariates like datetime attributes. However, I strongly agree with this statements from [felixdivo](https://github.com/felixdivo):

I'm totally with you that applying it to covariates as well should be possible. The reason I'm reluctant is that I'd want to find a unified solution for this accross all models / normalizations (also `use_reversible_instance_norm` (RINorm) and other models that use similar type of normalization).

> In my opinion, the darts implementation should be as close as possible to the original paper, only extending it where possible or fixing major flaws. At least that's what I would expect as a user of the library.

That's true but we also add additional features to models that didn't exist in the source paper to improve model performance (see RINorm for example, probabilistic support for all models, ...). So if we can find a neat way to generalize this logic, I'm all in.

> I would like to do so (as I need it to fix anyway for me personal to use)

Thanks, that's great to hear :) 🚀

**Comment by turbotimon** [comment]
@dennisbader I found another discussible thing in the implementation: With `const_init=True`, the weights in the layer will be const init, but the bias is still random. What do you think about that? I would suggest that the weights could be set to 0.0 in this case.  However, I'm not sure about that and the paper is unspecific in that case. The only thing I could find is *"we initialize the weights of the linear layers in DLinear as 1/L rather than random initialization."*.

P.S. Still looking forward to making a merge request (also because the `copy()` error currently is only fixed in the `shared_weights=False` part)

**Comment by turbotimon** [comment]
@dennisbader I finally found time to fix the nlinear #2757. Details on the fixes in the PR summary.

Some important remarks for the tests in [`test_dlinear_nlinear.py::TestDlinearNlinearModels.test_multivariate_and_covariates`](https://github.com/unit8co/darts/blob/42776790183bbe42411fc2dba3e1e8416f9263e8/darts/tests/models/forecasting/test_dlinear_nlinear.py#L132). I struggled for a while, but then decided to leave them untouched (for normalize=false) and implement others for the normalize=true:
- https://github.com/unit8co/darts/blob/42776790183bbe42411fc2dba3e1e8416f9263e8/darts/tests/models/forecasting/test_dlinear_nlinear.py#L207 looks like a error to me (using fut_cov1 for the past_covariates)? But anyway, the models are never tested with past covariates in this tests
- I'm questioning the test in general, as according to the paper, dlinear is most useful with trend and nlinear with distribution shifts in the data, but the test data used is stationary. Therefore, nlinear performs not good in these tests and dlinear is not tested for its intended usage/benefit.
